source/common/network/multi_connection_base_impl.cc

```diff
@@ -0,0 +1,591 @@
+#include "source/common/network/multi_connection_base_impl.h"
+
+#include <vector>
+
+namespace Envoy {
+namespace Network {
+
+MultiConnectionBaseImpl::MultiConnectionBaseImpl(Event::Dispatcher& dispatcher,
```

Comment:
This file was basically renamed from happy_eyeballs_connection_impl.cc, right? It's really unfortunate that git is failing to notice the copy and so the diff is showing up as complete new code. This makes it hard to see what's changed about it in this PR (other than the names :/). I have two thoughts for how to improve this though I'm not sure if either will work. Both will take some work.

First, I think we might be able to help git figure this out by having the first commit in this PR simply copy happy_eyeballs_connection_impl.{h,cc} => multi_connection_base_impl.{h,cc}. Then the second commit can make changes on top of that. You'd end up needing to do git push -f I think.

If that doesn't work, we could add a PR which simply does the copy but doesn't edit or use the new files. I'm sure that would work, though it would be slightly annoying. (Though better than losing the history)

Reply:
I've split the commit into two, it's still not very clear but we can now at least check the change introduced by the second commit with Github UI, thanks for the guide!

Reply:
I'm sure I'm just doing it wrong, but how I can I see the change to source/common/network/multi_connection_base_impl.cc which were made after copying it from source/common/network/happy_eyeballs_connection_impl.cc? (Sometimes the GH UI is really confusing!)

Reply:
I was able to check the changes for a single commit with link like [this](https://github.com/envoyproxy/envoy/pull/21539/commits/e10035dc214a852b6abe218bf8c44ec402d65a66), and different commit can be selected with "Files Changed" tab and then -> "Changes from all commits" dropdown list to select a particular commit.

Reply:
Ah! Thanks, I was having a hard time finding the right commit, I guess.!